File: ledger/ledger-utxo/src/main/java/net/corda/v5/ledger/utxo/BelongsToContract.java
package net.corda.v5.ledger.utxo;

import org.jetbrains.annotations.NotNull;

import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

/**
 * Indicates the {@link Contract} that the current state belongs to.
 */
@Target(ElementType.TYPE)
@Retention(RetentionPolicy.RUNTIME)
public @interface BelongsToContract {

    @NotNull
    Class<? extends Contract> value();
}

File: ledger/ledger-utxo/src/main/kotlin/net/corda/v5/ledger/utxo/BelongsToContract.kt
package net.corda.v5.ledger.utxo

import kotlin.reflect.KClass

/**
 * Indicates the [Contract] that the current state belongs to.
 */
@Retention(AnnotationRetention.RUNTIME)
@Target(AnnotationTarget.CLASS)
annotation class BelongsToContract(val value: KClass<out Contract>)
